### PR TITLE
(MODULES-8717) Move testing from task_helper to BoltSpec & Fix BoltSpec dependency issue

### PR DESCRIPTION
**Pull request by HelenCampbell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ This module is compatible with Puppet Enterprise and Puppet Bolt.
 
 * To run tasks with Puppet Enterprise, PE 2017.3 or later must be installed on the machine from which you are running task commands. Machines receiving task requests must be Puppet agents.
```
Comment (eimlav) [inline README.md]: Need to bump PE to 2018.1.

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+  inventory = hosts_to_inventory.merge('features' => ['puppet-agent'])
+  run_task(task_name, 'default', params, config: bolt_config, inventory: inventory)
+end
+
 RSpec.configure do |c|
   # Readable test descriptions
   c.formatter = :documentation
 
+  c.add_setting :module_path
+  c.module_path = File.join(base_dir, 'fixtures', 'modules')
```
Comment (eimlav) [inline spec/spec_helper_acceptance.rb]: These config variables are not needed when using the task_run helper.

File: spec/acceptance/windows_spec.rb [hunk 3]
```diff
@@ -2,8 +2,13 @@
 require 'spec_helper_acceptance'
 
 describe 'windows exec task', if: fact_on(default, 'osfamily') == 'windows' do
```
Comment (eimlav) [inline spec/acceptance/windows_spec.rb]: Probably worth using os[:family] here.

File: spec/acceptance/linux_spec.rb [hunk 4]
```diff
@@ -2,8 +2,13 @@
 require 'spec_helper_acceptance'
 
 describe 'linux exec task', unless: fact_on(default, 'osfamily') == 'windows' do
```
Comment (eimlav) [inline spec/acceptance/linux_spec.rb]: Probably worth using os[:family] here.

**Comment by pmcmaw** [comment]
Probably worth while documenting this so it can be carried out easily on our other modules. A quick explanation on strange steps may also help eg why do we need to update `.fixtures.yml`

**Comment by eimlav** [comment]
LGTM +1

**Comment by pmcmaw** [comment]
Do we want to remove puppet 5 unit testing, this is the only place we run these tests.

**Comment by eimlav** [comment]
@pmcmaw the task modules do not have any unit tests so we technically don't lose any test coverage. Keeping them also causes conflicts with the current approach we've used in implementing Bolt.

**Comment by pmcmaw** [comment]
@eimlav Thats such a shame. :-( Leaving the puppet5 testing in allows us to highlight things like gem dependencies with the puppet 5 gems etc.

File: spec/acceptance/windows_spec.rb (hunk 3 above)
Comment (HelenCampbell) [reply spec/acceptance/windows_spec.rb]: Going to leave these to get done as part of the acceptance audit

**Comment by Lavinia-Dan** [comment]
Ran it through adhoc, everything's green. LGTM too

**Comment by pmcmaw** [comment]
Merging.
Bolt 1.0 requires puppet 6 and greater. As we have no unit tests we aren't loosing coverage by removing puppet5 unit testing.

**Pull request merged**